Vagrantfile

```diff
@@ -42,6 +42,8 @@ ec2_region = "us-east-1"
 ec2_az = nil # Uses set by AWS
 ec2_ami = "ami-905730e8"
 ec2_instance_type = "m3.medium"
+ec2_spot_instance = ENV['SPOT_INSTANCE'] ? ENV['SPOT_INSTANCE'] == 'true' : true
+ec2_spot_max_price = ENV['SPOT_MAX_PRICE'] || "0.266"  # Price for on-demand instance for m3.xlarge
```

Comment:
I see we can override this via env variable, but having it fixed in code as a default seems like a bad baseline. Are we just always overriding this value in practice anyway?

Reply:
There is not much use for the env as it is normally not used, so removing it in favor of a sane default based on instance type. 

As for cherry-picking, as someone started testing 0.10.1, I think we can go back to there. There will be a merge conflict, but it's pretty straight forward / just due to Vagrant version change, which is the same version in 0.10.2 and that was tested to work.